### ISSUE TITLE
Proxy for Internet connection?

### ISSUE THREAD

**Issue by pefeigl** [ucla/moodle-mod_zoom#72]
Our server cannot create outbound connections. I do not want to globally set a proxy, as that would allow *every* moodle plugin to communicate with the outside world. Is there a way to set a proxy only for *this* plugin?

**Comment by scara** [comment]
Hi @pefeigl,
this plugin uses the [Moodle _curl_ wrapper class](https://github.com/moodle/moodle/blob/77d1c415025f99f1c36dae67aac7af11b8a34b50/lib/filelib.php#L2983) via [`mod_zoom_webservice ::_make_call()`](https://github.com/ucla/moodle-mod_zoom/blob/9e7a5d68865adbae52cd9aaa5f6a899def3a73d7/classes/webservice.php#L126) i.e. you should already be able to use the proxy settings of your Moodle instance: https://docs.moodle.org/38/en/HTTP#Web_proxy_settings.

HTH,
Matteo

**Comment by pefeigl** [comment]
> this plugin uses the [Moodle _curl_ wrapper class](https://github.com/moodle/moodle/blob/77d1c415025f99f1c36dae67aac7af11b8a34b50/lib/filelib.php#L2983) via [`mod_zoom_webservice ::_make_call()`](https://github.com/ucla/moodle-mod_zoom/blob/9e7a5d68865adbae52cd9aaa5f6a899def3a73d7/classes/webservice.php#L126) i.e. you should already be able to use the proxy settings of your Moodle instance: https://docs.moodle.org/38/en/HTTP#Web_proxy_settings.

I'd prefer not to give proxy access to the entire moodle, but *only* to
this one plugin :-/ Any way to achieve this?

Greetings, Peter

**Comment by scara** [comment]
Hi @pefeigl,
you can ack the plug-in code to add there those settings, directly in the function `_make_call()`: it's no common use though in the Moodle eco-system.

Unless @rlorenzo may consider this a valid use case and add these optional settings to allow this plug-in to overwrite Moodle _curl_ settings.

HTH,
Matteo

**Comment by rlorenzo** [comment]
We would accept any pull request You would have for this feature... but what plugins do you think would start talking to the world if you enable proxy? I only know of plugins that you configure server side settings that would then talk back to their respective servers.

**Comment by pefeigl** [comment]
@rlorenzo I've looked at the code, unfortunately moodle's curl class does not allow directly setting the proxy. So the only way to do this would be to save the global proxy settings, then change them, create the curl instance (you already do that in webservice.php), then change the settings back.

Would the following be acceptable to you:
I'll write a patch that adds a toggle for "use proxy only for zoom" (+ proxy settings) in the global mod settings. The code would then do the slightly hackish thing I described above.

If that's ok with you, I'll do a PR for this ;)
